Sources/Brave/Frontend/UserContent/UserScripts/Scripts_Dynamic/Scripts/Paged/FarblingProtectionScript.js

```diff
@@ -130,34 +130,45 @@ window.__firefox__.execute(function($) {
         // We need the original length so we can reference it (as we will change it)
         const plugins = window.navigator.plugins
         const originalPluginsLength = plugins.length
-
+        const pluginsPrototype = Object.getPrototypeOf(window.navigator.plugins)
+        
+        const fakePlugins = fakePluginData.map((pluginData) => {
+          return makeFakePlugin(pluginData)
+        })
+        
         // Adds a fake plugin for the given index on fakePluginData
-        const addPluginAtIndex = (newPlugin, index) => {
+        fakePlugins.forEach((newPlugin, index) => {
           const pluginPosition = originalPluginsLength + index
-          window.navigator.plugins[pluginPosition] = newPlugin
-          window.navigator.plugins[newPlugin.name] = newPlugin
-        }
-
-        for (const [index, pluginData] of fakePluginData.entries()) {
-          const newPlugin = makeFakePlugin(pluginData)
-          addPluginAtIndex(newPlugin, index)
-        }
-
-        // Adjust the length of the original plugin array
-        Reflect.defineProperty(window.navigator.plugins, 'length', {
-          value: originalPluginsLength + fakePluginData.length
+          pluginsPrototype[pluginPosition] = newPlugin
+          pluginsPrototype[newPlugin.name] = newPlugin
         })
-
-        // Patch `PluginArray.item(index)` function to return the correct item
-        // otherwise it returns `undefined`
-        const originalItemFunction = plugins.item
-        window.PluginArray.prototype.item = function (index) {
+        
+        // Farble the `item` method on the plugins array
+        const originalItem = window.navigator.plugins.item
+        pluginsPrototype.item = function () {
+          let index = arguments[0]
+          
           if (index < originalPluginsLength) {
-            return Reflect.apply(originalItemFunction, plugins, arguments)
+            return Reflect.apply(originalItem, this, arguments)
           } else {
-            return plugins[index]
+            const farbledIndex = index - originalPluginsLength
```

Comment:
If originalPluginsLength > index, you'd end up with a negative farbledIndex here.
Is it possible to have an originalPluginsLength > index?

Reply:
this is in an else of an if check which guarantees index is smaller than originalPluginsLength so the index may only be equal or less than original plugins length.

Reply:
I should also point out that if a user were to pass a negative value, this would fall into the if statement which then would get passed to the original method and handled as if unfarbled.